contracts/varatube/src/utils.rs

```diff
@@ -0,0 +1,132 @@
+use fungible_token_io::FTEvent;
+
+use crate::{msg, Actions, ActorId, Error, Period, Price, SubscriberData, VaraTube};
+
+impl VaraTube {
+    /// Add subscriber.
+    pub fn add_subscriber(&mut self, subscriber: &ActorId, data: SubscriberData) {
+        self.subscribers.insert(*subscriber, data);
+    }
+
+    /// Add pending subscription.
+    ///
+    /// Inserting `data` is actually currency id and subscription period.
+    pub fn add_pending_subscriber(&mut self, subscriber: &ActorId, data: (ActorId, Period)) {
+        let (currency_id, period) = data;
+        self.subscribers.insert(
+            *subscriber,
+            SubscriberData {
+                currency_id,
+                period,
+                subscription_start: None,
+                renewal_date: None,
+            },
+        );
+    }
+
+    /// Ger subscriber.
+    pub fn get_subscriber(&self, subscriber: &ActorId) -> Result<SubscriberData, Error> {
+        if let Some(subscriber_data) = self.subscribers.get(subscriber) {
+            Ok(*subscriber_data)
+        } else {
+            Err(Error::AccountDoesNotExist)
+        }
```

Comment:
Can we use `Option::ok_or` here?

```suggestion
        self.subscribers.get(subscriber)[.copied()].ok_or(Error::AccountDoesNotExist)
```